src/lib/i18n/i18n.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @typedef {import('./i18n.types.js').TranslationsMap} TranslationsMap
+ * @typedef {import('./i18n.types.js').Translations} Translations
+ * @typedef {import('./i18n.types.js').TranslateFunction} TranslateFunction
+ * @typedef {import('./i18n.types.js').Translated} Translated
+ */
+
+/**
+ * @param {K} key - The translation key
+ * @param {Object} [data] - The translation data
+ * @returns {Translated} - The translation
+ * @template {string} K
+ */
```

Comment:
issue: I don't think we need the `@template` anymore

:arrow_right: this change is available in my proposition branch: https://github.com/CleverCloud/clever-components/tree/typecheck-i18n--hubert